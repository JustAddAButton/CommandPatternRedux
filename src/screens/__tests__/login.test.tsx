// import * as React from 'react';
// import renderer from 'react-test-renderer';
// import {LoginScreen} from '../login';
// import {ILoginScreenProps} from '../ILoginScreenProps';
// import {Text} from 'react-native';
// import {LoginCommand} from '../../commands/LoginCommand';
// import {useCommand} from '../../commands/Command';
// import {renderHook} from '@testing-library/react-hooks';

// describe('Login Screen Tests', () => {
//   let testRenderer: renderer.ReactTestRenderer;

//   jest.mock('../../commands/LoginCommand', function () {
//     const {default: mockLoginCommand} = jest.requireActual(
//       '../../commands/LoginCommand',
//     );
//     mockLoginCommand.prototype.pending = false;
//     mockLoginCommand.prototype.error = 'An error has occurred.';
//     mockLoginCommand.prototype.canExecute = function () {
//       return true;
//     };

//     return mockLoginCommand;
//   });

//   beforeEach(() => {
//     const props: ILoginScreenProps = {
//       title: 'Testing Title',
//     };
//     testRenderer = renderer.create(<LoginScreen {...props} />);
//   });

//   it('Should show title', async () => {
//     const title = testRenderer.root.findAllByType(Text)[0];
//     expect(title.props.children).toBe('Testing Title');
//   });

//   it('Should show login error', async () => {
//     const title = testRenderer.root.findByProps({
//       children: 'An error has occurred',
//     });
//     expect(title.props.children).toBe('An error has occurred');
//   });

//   it('renders correctly', () => {
//     const tree = testRenderer.toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   test('should increment counter', () => {
//     var loginCommand = new LoginCommand();
//     loginCommand.pending = true;
//     loginCommand.error = 'An error has occurred';
//     jest.spyOn(loginCommand, 'canExecute').mockImplementation(() => true);

//     const {result} = renderHook(() => useCommand(() => loginCommand));
//     expect(result.current.pending).toBe(true);

//     const props: ILoginScreenProps = {
//       title: 'Testing Title',
//     };
//     const test = renderer.create(<LoginScreen {...props} />);
//     const title = test.root.findByProps({
//       children: 'An error has occurred',
//     });
//     expect(title.props.children).toBe('An error has occurred');
//   });
// });
