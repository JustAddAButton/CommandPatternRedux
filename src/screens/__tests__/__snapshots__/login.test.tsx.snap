// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen Tests renders correctly 1`] = `
<View>
  <View>
    <Text>
      Testing Title
    </Text>
  </View>
  <View>
    <Text>
      Email
    </Text>
    <TextInput
      onChangeText={[Function]}
      value=""
    />
  </View>
  <View>
    <Text>
      Password
    </Text>
    <TextInput
      onChangeText={[Function]}
      value=""
    />
  </View>
  <View>
    <Text>
      Remember Me
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      value={true}
    />
  </View>
  <View>
    <Text
      onPress={[Function]}
    />
  </View>
  <ActivityIndicator
    animating={false}
    size="large"
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {},
        ]
      }
    >
      <Text
        disabled={true}
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#cdcdcd",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
